cluster-autoscaler/utils/glogx/glogx.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package glogx
+
+import (
+	"github.com/golang/glog"
+)
+
+// UpTo decreases quota for logging and reports whether there was any left.
+// The returned value is a boolean of type glogx.Verbose.
+func UpTo(limit *int) glog.Verbose {
+	(*limit)--
+	return *limit >= 0
+}
+
+// Over reports whether quota for logging was exceeded.
+// The returned value is a boolean of type glogx.Verbose.
+func Over(limit *int) glog.Verbose {
+	return *limit < 0
+}
+
+// V calls V from glog and wraps the result into glogx.Verbose.
+func V(n glog.Level) Verbose {
+	return Verbose(glog.V(n))
+}
+
+// Verbose is a wrapper for glog.Verbose that implements UpTo and Over.
+type Verbose glog.Verbose
```

Comment:
This is too complicated. Can we simply create a struct that will allow to log at the given level only N times, with no extra int pointers etc.

Reply:
Do you mean that I should instead create an object with V() method, and use it instead of calling V() from package (something like Logger in Java?)

Reply:
refactored as discussed offline, PTAL